### PR TITLE
Update format.h

### PR DESCRIPTION
**Pull request by JurajX**
To prevent the following compiler message:
```
[...]/include/fmt/format.h:392:20: warning: conversion to 'long unsigned int' from 'int' may change the sign of the result [-Wsign-conversion]
  392 |     unsigned value[size];
      |                    ^~~~
```

### CODE REVIEW

File: include/fmt/format.h [hunk 1]
```diff
@@ -389,7 +389,7 @@ template <typename To, typename From, FMT_ENABLE_IF(sizeof(To) > sizeof(From))>
 inline auto bit_cast(const From& from) -> To {
   constexpr auto size = static_cast<int>(sizeof(From) / sizeof(unsigned));
   struct data_t {
-    unsigned value[size];
+    unsigned value[static_cast<size_t>(size)];
```
Comment (vitaut) [inline include/fmt/format.h:392]: Let's use to_unsigned instead of static_cast

**Review by vitaut — COMMENTED** [review]
Thank you for the PR.

File: include/fmt/format.h (hunk 1 above)
Comment (JurajX) [reply include/fmt/format.h]: Sure :)
In case I did something badly, feel free to fix it directly and close this PR. ;)

Comment (phprus) [reply include/fmt/format.h]: @vitaut ``to_unsigned`` is constexpr only since C++14.
I think simple ``static_cast`` is better here.

Comment (vitaut) [reply include/fmt/format.h]: Right. I switched back to cast and merged.

**Comment by vitaut** [comment]
Merged in https://github.com/fmtlib/fmt/commit/70dc3de053b6814c023faebbd21525aab575d25d, thanks.

**Pull request closed without merging**